saleor/graphql/meta/resolvers.py

```diff
@@ -106,5 +110,8 @@ def check_private_metadata_privilege(root: ModelWithMetadata, info):
 
 
 def resolve_private_metadata(root: ModelWithMetadata, info):
+    # Metadata for checkout uses separate model to store metadata
+    if isinstance(root, checkout_models.Checkout):
+        root = root.metadata
```

Comment:
Shouldn't we use dataloader here? We should also add a benchmark for checkout metadata or update the existing one.